csharp/src/Ice/OutgoingRequestFrame.cs

```diff
@@ -355,7 +362,14 @@ internal override void Finish()
             else
             {
                 Debug.Assert(Protocol == Protocol.Ice2);
-                ostr.WriteIce2RequestHeaderBody(Identity, Facet, Location, Operation, IsIdempotent);
+                ostr.WriteIce2RequestHeaderBody(Identity,
+                                                Facet,
+                                                Location,
+                                                Operation,
+                                                IsIdempotent,
+                                                // An infinite invocation timeout is encoded as 0
```

Comment:
I would use `-1`, which is the value for InfiniteTimeSpan.

Reply:
Fixed